--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -141,6 +141,11 @@ set_target_properties( E57Format
 		INTERPROCEDURAL_OPTIMIZATION_DEBUG OFF
 )
 
+if ( E57_BUILD_SHARED )
+	set_target_properties (E57Format PROPERTIES VERSION   ${PROJECT_VERSION})
+	set_target_properties (E57Format PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+endif()
+
 if( NOT DEFINED CMAKE_DEBUG_POSTFIX )
 	set_target_properties( E57Format
 		PROPERTIES
